src/models/ctx/update_profile.rs

```diff
@@ -82,10 +83,22 @@ pub fn update_profile<E: Env + 'static>(
                             .unwrap_or_else(|| profile_addon.to_owned())
                     })
                     .collect::<Vec<_>>();
-                let transport_urls = next_addons
+                let prev_transport_urls = profile
+                    .addons
+                    .iter()
+                    .map(|addon| &addon.transport_url)
+                    .cloned()
+                    .collect::<HashSet<_>>();
+                let next_transport_urls = next_addons
                     .iter()
                     .map(|addon| &addon.transport_url)
                     .cloned()
+                    .collect::<HashSet<_>>();
+                let added_transport_urls = &next_transport_urls - &prev_transport_urls;
+                let removed_transport_urls = &prev_transport_urls - &next_transport_urls;
```

Comment:
Why do we need to add the removed `transport_url`s to the `AddonsPulledFromAPI` event?